### PR TITLE
Footer button

### PR DESCRIPTION
**Pull request by fondation451**
Issue: https://github.com/fondation451/label/issues/181

## Changes
- The button of the footer are working (draft and validate)
- It highlights the fact that we need a really better mutualization at the graphQL level. I removed some useless code that I made for the mutation. The mutation are handwritten. In the next PR, I will try to built these schema automatically.

### CODE REVIEW

File: packages/generic/client/src/pages/Home/DocumentAnnotator/DocumentPanel/DocumentPanelFooter.tsx [hunk 1]
```diff
@@ -19,11 +38,11 @@ function DocumentPanelFooter(props: { annotatorStateHandler: annotatorStateHandl
       </LayoutGrid>
       <LayoutGrid item>
         {/* eslint-disable-next-line @typescript-eslint/no-empty-function */}
```
Comment (BenoitSerrano) [inline packages/generic/client/src/pages/Home/DocumentAnnotator/DocumentPanel/DocumentPanelFooter.tsx:40]: plus besoin du /* eslint disabled */

File: packages/generic/client/src/pages/Home/DocumentAnnotator/DocumentPanel/DocumentPanelFooter.tsx [hunk 2]
```diff
@@ -19,11 +38,11 @@ function DocumentPanelFooter(props: { annotatorStateHandler: annotatorStateHandl
...
-        <Button onClick={() => {}} color="default" iconName="save" style={styles.saveDraftButton}>
+        <Button onClick={saveDraft} color="default" iconName="save" style={styles.saveDraftButton}>
           {wordings.saveDraft}
         </Button>
         {/* eslint-disable-next-line @typescript-eslint/no-empty-function */}
```
Comment (BenoitSerrano) [inline packages/generic/client/src/pages/Home/DocumentAnnotator/DocumentPanel/DocumentPanelFooter.tsx:44]: plus besoin du /* eslint disabled */

File: packages/generic/client/src/pages/Home/DocumentAnnotator/DocumentPanel/DocumentPanelFooter.tsx [hunk 3]
```diff
@@ -41,4 +60,22 @@ function DocumentPanelFooter(props: { annotatorStateHandler: annotatorStateHandl
...
+  }
+
+  function saveAnnotationsAndUpdateAssignationStatus(status: assignationStatusType) {
+    saveAnnotations({
+      variables: {
+        documentIdString: annotatorState.document._id,
+        fetchedGraphQLAnnotations: annotatorState.annotations,
+      },
+    });
+    updateAssignationStatus({ variables: { documentIdString: annotatorState.document._id, statusString: status } });
```
Comment (BenoitSerrano) [inline packages/generic/client/src/pages/Home/DocumentAnnotator/DocumentPanel/DocumentPanelFooter.tsx:79]: pourquoi statusString ?

**Review by BenoitSerrano — APPROVED** [review]

File: packages/generic/client/src/pages/Home/DocumentAnnotator/DocumentPanel/DocumentPanelFooter.tsx (hunk 2 above)
Comment (fondation451) [reply packages/generic/client/src/pages/Home/DocumentAnnotator/DocumentPanel/DocumentPanelFooter.tsx]: @BenoitSerrano done 
0690f0a

File: packages/generic/client/src/pages/Home/DocumentAnnotator/DocumentPanel/DocumentPanelFooter.tsx (hunk 1 above)
Comment (fondation451) [reply packages/generic/client/src/pages/Home/DocumentAnnotator/DocumentPanel/DocumentPanelFooter.tsx]: @BenoitSerrano Done 
0690f0a

**Pull request merged**